### PR TITLE
Fixes #25483 - keep false facts

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

**Comment by ares** [comment]
we ignore all incoming facts with value false, becase there was a check for `value.present?`. `false` value is never present :-)

File: app/services/structured_fact_importer.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class StructuredFactImporter < FactImporter
   def normalize(facts)
     # Remove empty values first, so nil facts added by flatten_composite imply compose
-    facts.keep_if { |k, v| v.present? }
+    facts.keep_if { |k, v| !v.nil? || (v.is_a?(String) && !v.empty?) }
```
Comment (ekohl) [inline app/services/structured_fact_importer.rb]: Maybe `delete_if { |k, v| v.nil? || (v.is_a?(String) && v.empty?) }` is easier to read

Comment (lzap) [reply app/services/structured_fact_importer.rb]: Sure, can you rebase please?

**Review by lzap — CHANGES_REQUESTED** [review]
This needs a test.

**Comment by ares** [comment]
modified the condition, rebased and added a test

**Review by ekohl — DISMISSED** [review]

**Review by lzap — DISMISSED** [review]

**Comment by lzap** [comment]
Does not seem to be related but we need to see tests all green.

```
Offenses:

app/helpers/operatingsystems_helper.rb:9:3: C: Metrics/CyclomaticComplexity: Cyclomatic complexity for icon is too high. [24/23]
  def icon(record, opts = {}) ...
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^

2229 files inspected, 1 offense detected
```

**Comment by ares** [comment]
@lzap that's unrelated, being discussed at https://github.com/theforeman/foreman/pull/6356

**Comment by mmoll** [comment]
ok to test

**Review by mmoll — APPROVED** [review]

**Comment by mmoll** [comment]
GH-6355 already fixed the rubocop failure

**Comment by mmoll** [comment]
merged, díky @ares!

**Pull request merged**